zetaclient/types/event.go

```diff
@@ -1,7 +1,31 @@
 package types
 
 import (
+	"bytes"
+	"encoding/hex"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
 	"github.com/zeta-chain/node/pkg/coin"
+	"github.com/zeta-chain/node/pkg/constant"
+	"github.com/zeta-chain/node/pkg/crypto"
+	"github.com/zeta-chain/node/pkg/memo"
+	"github.com/zeta-chain/node/zetaclient/config"
+)
+
+// InboundProcessability is an enum representing the processability of an inbound
+type InboundProcessability int
+
+const (
+	// InboundProcessabilityGood represents a processable inbound
+	InboundProcessabilityGood InboundProcessability = iota
```

Comment:
let's not introduce net-new concepts globally and keep this in BTC implementation only (as it was). Related: https://github.com/zeta-chain/node/issues/2977

Reply:
It is used by both SOL and BTC chain.